### PR TITLE
JAMES-2957 dlp.Dlp matcher cannot parse emails containing attachments

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

**Comment by trantienduchn** [comment]
test this please

**Comment by trantienduchn** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

File: server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/dlp/DlpDomainRules.java [hunk 1]
```diff
@@ -106,16 +106,22 @@ public boolean doTest(Mail mail) throws MessagingException, IOException {
                 if (content instanceof String) {
                     return Stream.of((String) content);
                 }
+
+                return extractContentsComplexType(content)
```
Comment (mbaechler) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/dlp/DlpDomainRules.java]: First, I'm not sure what is tail recursive ...
Second, AFAIK Java doesn't support tail recursion optimization ...

**Pull request closed without merging**